node/network/protocol/src/request_response/request.rs

```diff
@@ -281,6 +278,98 @@ where
 	}
 }
 
+/// Typed variant of [`netconfig::OutgoingResponse`].
+///
+/// Responses to `IncomingRequest`s.
+pub struct OutgoingResponse<Response> {
+	/// The payload of the response.
+	pub result: Result<Response, ()>,
+
+	/// Reputation changes accrued while handling the request. To be applied to the reputation of
+	/// the peer sending the request.
```

Comment:
Does this have to be a vec? So far (at least I) went with rep changes based on the first _violation_ detected. Not sure if this simplification can be applied here as well.

Reply:
Not sure this is useful at all, I was just mimicking the underlying low level substrate API. Argument against that change would be consistency across APIs - can't think of any other.

Reply:
Unless we have a good reason, we should cut the `vec![]`s down to one place at the API boundary imho.

Reply:
I have a "good" reason. If it is not a `Vec` it has to be an `Option`, so we still need to wrap the reputation. To me having to do a `vec![reputation]` is not really any worse than doing `Some(reputation)`, but at the same time the vec is slightly more expressive.

Reply:
What's the concern with a `Vec`? Allocation performance? Could be `SmallVec<[UnifiedReputationChange; 1]>`